src/command_modules/azure-cli-rdbms/tests/test_rdbms_commands.py

```diff
@@ -0,0 +1,377 @@
+
+        # self.cmd('{} db delete -g {} -s {} -n {} --yes'
+        #         .format(engine, rg, server, database_name),
+        #         checks=[NoneCheck()])
+
+    def _test_configuration_mgmt(self, resource_group, server, engine):
+        rg = resource_group
+        if engine == 'mysql':
+            config_name = 'event_scheduler'
+            default_value = 'OFF'
+            new_value = 'ON'
+        else:
+            config_name = 'array_nulls'
+            default_value = 'on'
+            new_value = 'off'
+
+        # test show configuration
+        self.cmd('{} server configuration show --name {} -g {} --server {}'
+                 .format(engine, config_name, rg, server),
+                 checks=[
+                     JMESPathCheck('name', config_name),
+                     JMESPathCheck('value', default_value),
+                     JMESPathCheck('source', 'system-default')])
+
+        # test update configuration
+        self.cmd('{} server configuration set -n {} -g {} -s {} --value {}'
+                 .format(engine, config_name, rg, server, new_value),
+                 checks=[
+                     JMESPathCheck('name', config_name),
+                     JMESPathCheck('value', new_value),
+                     JMESPathCheck('source', 'user-override')])
+
+        self.cmd('{} server configuration set -n {} -g {} -s {}'
+                 .format(engine, config_name, rg, server),
+                 checks=[
+                     JMESPathCheck('name', config_name),
+                     JMESPathCheck('value', default_value)])
+
+        # test list configurations
+        self.cmd('{} server configuration list -g {} -s {}'
+                 .format(engine, rg, server),
+                 checks=[JMESPathCheck('type(@)', 'array')])
+
+    def _test_log_file_mgmt(self, resource_group, server, engine):
+        rg = resource_group
+
+        if engine == 'mysql':
+            config_name = 'slow_query_log'
+            new_value = 'ON'
+
+            # test update configuration
+            self.cmd('{} server configuration set -n {} -g {} -s {} --value {}'
+                     .format(engine, config_name, rg, server, new_value),
+                     checks=[
+                         JMESPathCheck('name', config_name),
+                         JMESPathCheck('value', new_value)])
+
+        # test list log files
+        result = self.cmd('{} server-logs list -g {} -s {}'
+                          .format(engine, rg, server),
+                          checks=[
+                              JMESPathCheck('length(@)', 1),
+                              JMESPathCheck('type(@)', 'array')]).get_output_in_json()
+
+        # test download file
+        file_name = result[0]['name']
+        self.cmd('{} server-logs download -g {} -s {} --name {}'
+                 .format(engine, rg, server, file_name),
+                 checks=[NoneCheck()])
+
+        import os
+        os.remove(file_name)
```

Comment:
This will not guarantee the file is deleted in the case when the previous assertion failed. Please use `self.addCleanup` to register clean up code. https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup